opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/SpanShim.java

```diff
@@ -214,10 +238,40 @@ static Attributes convertToAttributes(Map<String, ?> fields) {
       } else if (value instanceof Boolean) {
         attributesBuilder.put(booleanKey(key), (Boolean) value);
       } else {
-        attributesBuilder.put(stringKey(key), value.toString());
+        AttributeKey<String> attributeKey = null;
+        if (isError && !isRecordingException) {
+          if (key.equals(Fields.ERROR_KIND)) {
+            attributeKey = SemanticAttributes.EXCEPTION_TYPE;
+          } else if (key.equals(Fields.MESSAGE)) {
+            attributeKey = SemanticAttributes.EXCEPTION_MESSAGE;
+          } else if (key.equals(Fields.STACK)) {
+            attributeKey = SemanticAttributes.EXCEPTION_STACKTRACE;
+          }
+        }
+        if (!isRecordingException
```

Comment:
Let's make the code match the spec closer

`along the rest of the specified key/value pair`

This appears to say that we record exception with the value of ERROR_OBJECT, and everything else ("the rest") is copied as is.

So I think the logic can be added above the complex if statement as just

```java
if (isRecordingException && key.equals(Fields.ERROR_OBJECT) {
  // Already recorded as the exception itself so don't add as attribute.
  continue;
}
```

Then I think this is just `if (attributeKey == null)`

Reply:
updated